src/proof/nova.rs

```diff
@@ -340,16 +340,16 @@ impl<'a: 'b, 'b, C: Coprocessor<S1>> Proof<'a, C> {
         &self,
         pp: &PublicParams<'_, C>,
         num_steps: usize,
-        z0: Vec<S1>,
+        z0: &[S1],
         zi: &[S1],
     ) -> Result<bool, NovaError> {
         let (z0_primary, zi_primary) = (z0, zi);
         let z0_secondary = Self::z0_secondary();
         let zi_secondary = z0_secondary.clone();
 
         let (zi_primary_verified, zi_secondary_verified) = match self {
-            Self::Recursive(p) => p.verify(&pp.pp, num_steps, &z0_primary, &z0_secondary),
-            Self::Compressed(p) => p.verify(&pp.vk, num_steps, z0_primary, z0_secondary),
+            Self::Recursive(p) => p.verify(&pp.pp, num_steps, z0_primary, &z0_secondary),
+            Self::Compressed(p) => p.verify(&pp.vk, num_steps, z0_primary.to_vec(), z0_secondary),
```

Comment:
This masks a clone. See [C-CALLER-CONTROL](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control)

Reply:
The clone may never happen (in the `Self::Recursive` case)